compiler/modules/types/list.bal

```diff
@@ -53,6 +53,46 @@ public function listAtomicTypeAllMemberTypes(ListAtomicType atomicType) returns
     return [ranges, types];
 }
 
+class MemoListBddCache {
+    *BddCache;
+    private final MemoBddCache memoCache;
+    private final Context cx;
+
+    function init(Context cx) {
+        self.cx = cx;
+        self.memoCache = new(cx.listMemo);
+    }
+    
+    isolated function get(Bdd bdd) returns Bdd {
+        return self.memoCache.get(bdd);
+    }
+
+    isolated function simpleIntersect(Bdd b1, Bdd b2) returns Bdd? {
+        if b1 !is BddNode || b2 !is BddNode ||
```

Comment:
Separate out a function that converts a Bdd to an `Atom?` (non-nil in case it's a single BddNode with that atom)

Reply:
function removed